src/Donors/ServiceProvider.php

```diff
@@ -65,4 +69,22 @@ private function addCustomFieldsToDonorDetails()
             echo (new DonorDetailsController())->show($donor);
         });
     }
+
+    /**
+     * Hook into the donor creation process to ensure that donors are also users.
+     * @unreleased
+     */
+    protected function enforceDonorsAsUsers()
+    {
+        add_action('givewp_donate_controller_donor_created', function(Donor $donor, $formId) {
+            if(!$donor->userId) {
+
+                give(CreateUserFromDonor::class)->__invoke($donor);
+
+                if(DonationForm::find($formId)->settings->registrationNotification) {
```

Comment:
Can you run reformat on these files? Some of the formatting looks suspect 🧐

Reply:
Resolved, see https://github.com/impress-org/givewp/pull/7103/commits/c159196b4fd1165c542a8bd77b00bd2eed31cd0a.